### ISSUE TITLE
NullReferenceException ->  Called activePendingPiecesCoroutine

### ISSUE THREAD

**Issue by Lombrar** [zolantris/ValheimMods#50]
### What happened?

At some point playing at the server, my log gets spammed with the following messages:

[Debug  :ValheimRAFT.MoveableBaseRootComponent] Called activePendingPiecesCoroutine, pendingPieces count: 2
[Debug  :ValheimRAFT.MoveableBaseRootComponent+<ActivatePendingPieces>d__64] Ship Size calc is: m_bounds Center: (0.00, 0.00, 0.00), Extents: (0.00, 0.00, 0.00) bounds size (0.00, 0.00, 0.00)
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
ZNetScene.RemoveObjects (System.Collections.Generic.List`1[T] currentNearObjects, System.Collections.Generic.List`1[T] currentDistantObjects) (at <dbd2a6fbcde9498cadcacfb37ef883e3>:0)
(wrapper dynamic-method) ZNetScene.DMD<ZNetScene::CreateDestroyObjects>(ZNetScene)
(wrapper dynamic-method) ZNetScene.DMD<ZNetScene::Update>(ZNetScene)

[Debug  :ValheimRAFT.Patches.ValheimRAFT_Patch] Update with HaveControllingPlayer FLAG, False
[Debug  :ValheimRAFT.Patches.ValheimRAFT_Patch] Update with HaveControllingPlayer FLAG, False
[Debug  :ValheimRAFT.Patches.ValheimRAFT_Patch] Update with HaveControllingPlayer FLAG, False
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
ZNetScene.RemoveObjects (System.Collections.Generic.List`1[T] currentNearObjects, System.Collections.Generic.List`1[T] currentDistantObjects) (at <dbd2a6fbcde9498cadcacfb37ef883e3>:0)
(wrapper dynamic-method) ZNetScene.DMD<ZNetScene::CreateDestroyObjects>(ZNetScene)
(wrapper dynamic-method) ZNetScene.DMD<ZNetScene::Update>(ZNetScene)

[Debug  :ValheimRAFT.Patches.ValheimRAFT_Patch] Update with HaveControllingPlayer FLAG, False
[Debug  :ValheimRAFT.Patches.ValheimRAFT_Patch] Update with HaveControllingPlayer FLAG, False
[Debug  :ValheimRAFT.Patches.ValheimRAFT_Patch] Update with HaveControllingPlayer FLAG, False
[Debug  :ValheimRAFT.Patches.ValheimRAFT_Patch] Update with HaveControllingPlayer FLAG, False
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
ZNetScene.RemoveObjects (System.Collections.Generic.List`1[T] currentNearObjects, System.Collections.Generic.List`1[T] currentDistantObjects) (at <dbd2a6fbcde9498cadcacfb37ef883e3>:0)
(wrapper dynamic-method) ZNetScene.DMD<ZNetScene::CreateDestroyObjects>(ZNetScene)
(wrapper dynamic-method) ZNetScene.DMD<ZNetScene::Update>(ZNetScene)

[Debug  :ValheimRAFT.Patches.ValheimRAFT_Patch] Update with HaveControllingPlayer FLAG, False
[Debug  :ValheimRAFT.Patches.ValheimRAFT_Patch] Update with HaveControllingPlayer FLAG, False
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
ZNetScene.RemoveObjects (System.Collections.Generic.List`1[T] currentNearObjects, System.Collections.Generic.List`1[T] currentDistantObjects) (at <dbd2a6fbcde9498cadcacfb37ef883e3>:0)
(wrapper dynamic-method) ZNetScene.DMD<ZNetScene::CreateDestroyObjects>(ZNetScene)
(wrapper dynamic-method) ZNetScene.DMD<ZNetScene::Update>(ZNetScene)

[Debug  :ValheimRAFT.Patches.ValheimRAFT_Patch] Update with HaveControllingPlayer FLAG, False
[Debug  :ValheimRAFT.Patches.ValheimRAFT_Patch] Update with HaveControllingPlayer FLAG, False
[Debug  :ValheimRAFT.Patches.ValheimRAFT_Patch] Update with HaveControllingPlayer FLAG, False
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
ZNetScene.RemoveObjects (System.Collections.Generic.List`1[T] currentNearObjects, System.Collections.Generic.List`1[T] currentDistantObjects) (at <dbd2a6fbcde9498cadcacfb37ef883e3>:0)
(wrapper dynamic-method) ZNetScene.DMD<ZNetScene::CreateDestroyObjects>(ZNetScene)
(wrapper dynamic-method) ZNetScene.DMD<ZNetScene::Update>(ZNetScene)

[Debug  :ValheimRAFT.Patches.ValheimRAFT_Patch] Update with HaveControllingPlayer FLAG, False
[Debug  :ValheimRAFT.Patches.ValheimRAFT_Patch] Update with HaveControllingPlayer FLAG, False
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
ZNetScene.RemoveObjects (System.Collections.Generic.List`1[T] currentNearObjects, System.Collections.Generic.List`1[T] currentDistantObjects) (at <dbd2a6fbcde9498cadcacfb37ef883e3>:0)
(wrapper dynamic-method) ZNetScene.DMD<ZNetScene::CreateDestroyObjects>(ZNetScene)
(wrapper dynamic-method) ZNetScene.DMD<ZNetScene::Update>(ZNetScene)

[Debug  :ValheimRAFT.Patches.ValheimRAFT_Patch] Update with HaveControllingPlayer FLAG, False
[Debug  :ValheimRAFT.Patches.ValheimRAFT_Patch] Update with HaveControllingPlayer FLAG, False
[Debug  :ValheimRAFT.Patches.ValheimRAFT_Patch] Update with HaveControllingPlayer FLAG, False
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
ZNetScene.RemoveObjects (System.Collections.Generic.List`1[T] currentNearObjects, System.Collections.Generic.List`1[T] currentDistantObjects) (at <dbd2a6fbcde9498cadcacfb37ef883e3>:0)
(wrapper dynamic-method) ZNetScene.DMD<ZNetScene::CreateDestroyObjects>(ZNetScene)
(wrapper dynamic-method) ZNetScene.DMD<ZNetScene::Update>(ZNetScene)

and so on ...

the Debug rows are from another mod, helping me to find the root.
I have ValheimRaft up to date. What could it be?

### Steps to reproduce

Ingame playing for a while. Most likely to happen in meadows for some reason.

### Log outputs

```txt
[Debug  :ValheimRAFT.MoveableBaseRootComponent] Called activePendingPiecesCoroutine, pendingPieces count: 2
[Debug  :ValheimRAFT.MoveableBaseRootComponent+<ActivatePendingPieces>d__64] Ship Size calc is: m_bounds Center: (0.00, 0.00, 0.00), Extents: (0.00, 0.00, 0.00) bounds size (0.00, 0.00, 0.00)
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
ZNetScene.RemoveObjects (System.Collections.Generic.List`1[T] currentNearObjects, System.Collections.Generic.List`1[T] currentDistantObjects) (at <dbd2a6fbcde9498cadcacfb37ef883e3>:0)
(wrapper dynamic-method) ZNetScene.DMD<ZNetScene::CreateDestroyObjects>(ZNetScene)
(wrapper dynamic-method) ZNetScene.DMD<ZNetScene::Update>(ZNetScene)

[Debug  :ValheimRAFT.Patches.ValheimRAFT_Patch] Update with HaveControllingPlayer FLAG, False
[Debug  :ValheimRAFT.Patches.ValheimRAFT_Patch] Update with HaveControllingPlayer FLAG, False
[Debug  :ValheimRAFT.Patches.ValheimRAFT_Patch] Update with HaveControllingPlayer FLAG, False
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
ZNetScene.RemoveObjects (System.Collections.Generic.List`1[T] currentNearObjects, System.Collections.Generic.List`1[T] currentDistantObjects) (at <dbd2a6fbcde9498cadcacfb37ef883e3>:0)
(wrapper dynamic-method) ZNetScene.DMD<ZNetScene::CreateDestroyObjects>(ZNetScene)
(wrapper dynamic-method) ZNetScene.DMD<ZNetScene::Update>(ZNetScene)

[Debug  :ValheimRAFT.Patches.ValheimRAFT_Patch] Update with HaveControllingPlayer FLAG, False
[Debug  :ValheimRAFT.Patches.ValheimRAFT_Patch] Update with HaveControllingPlayer FLAG, False
[Debug  :ValheimRAFT.Patches.ValheimRAFT_Patch] Update with HaveControllingPlayer FLAG, False
[Debug  :ValheimRAFT.Patches.ValheimRAFT_Patch] Update with HaveControllingPlayer FLAG, False
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
ZNetScene.RemoveObjects (System.Collections.Generic.List`1[T] currentNearObjects, System.Collections.Generic.List`1[T] currentDistantObjects) (at <dbd2a6fbcde9498cadcacfb37ef883e3>:0)
(wrapper dynamic-method) ZNetScene.DMD<ZNetScene::CreateDestroyObjects>(ZNetScene)
(wrapper dynamic-method) ZNetScene.DMD<ZNetScene::Update>(ZNetScene)
```

### Environment

```markdown
OS:
- **OS**: Windows 11 | Ubuntu 20.04
- **Graphics**: DX11 (or Vulkan)
Mods:
- **BepInEx**: 5.4.2202
- Jotunn 2.15.2
- **ValheimRAFT**: Major.Minor.Patch
[Info   :   BepInEx] Loading [MadHatter 2.1.7]
[Info   :   BepInEx] Loading [OdinsKingdom 1.2.17]
[Info   :   BepInEx] Loading [Blacksmithing 1.2.5]
[Info   :   BepInEx] Loading [Cooking 1.1.15]
[Info   :   BepInEx] Loading [Dual Wield 1.0.9]
[Info   :   BepInEx] Loading [Farming 2.1.12]
[Info   :   BepInEx] Loading [Foraging 1.0.8]
[Info   :   BepInEx] Loading [HildirsQuest 1.0.2]
[Info   :   BepInEx] Loading [Lumberjacking 1.0.5]
[Info   :   BepInEx] Loading [Mining 1.1.5]
[Info   :   BepInEx] Loading [Network 1.0.2]
[Info   :   BepInEx] Loading [PackHorse 1.0.3]
[Info   :   BepInEx] Loading [Sailing 1.1.7]
[Info   :   BepInEx] Loading [Tenacity 1.0.3]
[Info   :   BepInEx] Loading [Display BepInEx Info In-Game 2.0.2]
[Info   :   BepInEx] Loading [PungusSouls 0.0.7]
[Info   :   BepInEx] Loading [SearsCatalog 1.4.0]
[Info   :   BepInEx] Loading [MoreVanillaBuildPrefabs 1.0.3]
[Debug  :MoreVanillaBuildPrefabs] Configuration manager found, hooking DisplayingWindowChanged
[Info   :   BepInEx] Loading [Server Devcommands 1.67]
[Info   :   BepInEx] Loading [SouthsilArmor 3.0.4]
[Info   :   BepInEx] Loading [AutomaticFuel 1.3.9]
[Info   :   BepInEx] Loading [TreesReborn 1.0.8]
[Info   :   BepInEx] Loading [Warfare 1.6.2]
[Info   :   BepInEx] Loading [Armory 1.1.6]
[Info   :   BepInEx] Loading [Monstrum 1.3.2]
[Info   :   BepInEx] Loading [WarfareFireAndIce 1.1.1]
[Info   :   BepInEx] Loading [MonstrumAshlands 1.1.4]
[Info   :   BepInEx] Loading [MonstrumDeepNorth 1.1.3]
[Info   :   BepInEx] Loading [Upgrade World 1.46]
[Info   :   BepInEx] Loading [Vortex Worlds 1.0.1]
```

**Comment by zolantris** [comment]
My guess is you have a conflict between this mod and the sailing mod. What happens if you remove sailing?

**Comment by Lombrar** [comment]
The error still occurs. When deactivating valheimRaft, the error is removed.

**Comment by zolantris** [comment]
Any luck with this error going away in 1.6.12?

**Comment by Lombrar** [comment]
I am sorry, i did not try again after your new patch, because I wanted to continue to mod with my friend. I will try it out tho within the next days.

**Comment by Lombrar** [comment]
Update: We tried the new version, but the error still exists. My guess would be, that we have some Sail Cornerparts left in our world, that the game cant delete itself. At least, it is what the log says.

**Comment by zolantris** [comment]
Unity Explorer will let you manually delete those items. They are called SailComponent. You can find it on thunderstore. It's a powerful too so be careful to only delete that object.

**Comment by zolantris** [comment]
Send me your latest save. I'll look at it too. I can try to add guards to prevent this problem

**Comment by Lombrar** [comment]
Sure, but as a reminder, we are using over 60+ plugins on our save. So far, no major incompabilities w/o ValheimRaft:

https://www.file-upload.net/download-15257486/save.rar.html

I will try your mentioned tool, thanks!

**Comment by zolantris** [comment]
Do you have a thunderstore mod file I can download. There is an option to export a mod list file.

**Comment by Lombrar** [comment]
Apparently not. If it helps, I screenshotted my modlist and uploaded it here:

https://www.file-upload.net/download-15257516/Desktop.rar.html

**Comment by Lombrar** [comment]
I installed UE, but i can't find any SailComponents. Do I have to activate valheimRAFT to see them?

**Comment by Lombrar** [comment]
![image](https://github.com/zolantris/ValheimMods/assets/79539291/f9156d6d-c858-46c1-8e79-d2688b319bba)

This error still persists, but it isnt spamming the console anymore, hence lagging the game.

**Comment by ma11088** [comment]
This is the same issue I am having, but I have found, at least for me, it is exclusive to running on a server. I can run the same world file locally without issue. 

removing ValheimRaft is the only way my server can run. No conflicts, otherwise.

**Comment by ma11088** [comment]
Using Unity Explorer, I found the ValheimRAFT.SailComponent and removed it. All issues are gone; everything is working beautifully on my server now.

**Comment by Lombrar** [comment]
Can you tell me where to look for it? I used the Object search and typed "SailComponent" but nothing showed up. Only when searching with "Sail" stuff pops up

**Comment by ma11088** [comment]
> Can you tell me where to look for it? I used the Object search and typed "SailComponent" but nothing showed up. Only when searching with "Sail" stuff pops up

Just had to restore an old world to replicate the problem, but I am positive that this is what I did, keeping in mind that I did this locally and then uploaded it to my server once I confirmed it was fixed:

Enter the world. Check with Bepin to see your error text is appearing, as it would be.

Object Explorer > Search UnityObject > Class Filter: (I just type sail and then select the following) "valheimraft.sailcomponent" > Select MBSail > in the window that pops up, select "Inspect Game Object" > Click "Destroy"

Save the world, log out or exit, start it up again and confirm that your world isn't giving you the errors, reupload to server.
